Engines/Wine/Engine/Versions/script.js

```diff
@@ -122,15 +123,24 @@ module.getAvailableVersions = function (wizard) {
 
 
 module.getLatestStableVersion = function (wizard, architecture) {
-    return getLatestVersion(wizard, "upstream-linux-" + architecture, /^\d+\.0(\.\d+)?$/);
+    const os = operatingSystemFetcher.fetchCurrentOperationSystem().getWinePackage()
+    const arch = operatingSystemFetcher.fetchCurrentOperationSystem().getWinePackage() === "darwin" ? "x86on64" : "x86";
```

Comment:
Use the `architecture` parameter. You've already fixed the default `_wineArchitecture` in `QuickScript`, so it should work without further changes.